src/transformers/training_args.py

```diff
@@ -1969,6 +1982,9 @@ def __post_init__(self):
             if self.fsdp_config["xla_fsdp_grad_ckpt"]:
                 warnings.warn("`--xla_fsdp_grad_ckpt` is useful only when `--xla` is set to true.")
 
+        if self.tp_size > 1:
+            os.environ["ACCELERATE_USE_TP"] = "true"
+            os.environ["TP_SIZE"] = str(self.tp_size)
```

Comment:
During here we should also guard the `accelerate` version required to use `tp`, can you add this please? :)

Reply:
Have added this guard. Thank you for noticing it.

Reply:
Can we push for a merge? Thank you.